File: app/src/main/java/com/koresuniku/wishmaster/presenter/LoadDataView.java
package com.koresuniku.wishmaster.presenter;

import android.app.Activity;
import android.widget.ProgressBar;

import com.koresuniku.wishmaster.http.IBaseJsonSchema;

import java.util.List;

public interface LoadDataView<T extends IBaseJsonSchema> {

    void onDataLoaded(List<T> schema);

    Activity getActivity();

    void showProgressBar();
}

File: app/src/main/java/com/koresuniku/wishmaster/presenter/view_interface/LoadDataView.kt
package com.koresuniku.wishmaster.presenter.view_interface

import android.app.Activity
import android.widget.ProgressBar

import com.koresuniku.wishmaster.http.IBaseJsonSchema

interface LoadDataView {

    fun onDataLoaded(schema: List<IBaseJsonSchema>)

    val activity: Activity

    fun showProgressBar()
}
